oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java

```diff
@@ -1264,34 +1133,22 @@ private boolean loadItems() {
             try {
                 currentThread().setContextClassLoader(getClass().getClassLoader());
 
-                CloudBlobContainer container = azureBlobContainerProvider.getBlobContainer();
-                if (!firstCall && (resultContinuation == null || !resultContinuation.hasContinuation())) {
-                    LOG.trace("No more records in container. containerName={}", container);
+                if (!firstCall) {
```

Comment:
The pagination has been removed. In this case filling up a list in one go will potentially cause OOM. 
The result object is a PagedIterable and getting an iterator() ans delegating to it should be much more simpler.

Reply:
Ref https://learn.microsoft.com/en-us/java/api/com.azure.storage.blob.models?view=azure-java-stable